core/blockchain.go

```diff
@@ -355,6 +356,21 @@ func (bc *BlockChain) GetVMConfig() *vm.Config {
 	return &bc.vmConfig
 }
 
+// NewEVMCaller retuns the EVMCaller factory of the blockchain
+func (bc *BlockChain) NewEVMCaller(header *types.Header, state vm.StateDB) vm.EVMCaller {
+	return vmcontext.NewEVMCaller(bc, header, state)
+}
+
+// NewEVMCaller retuns the EVMCaller factory of the blockchain
+func (bc *BlockChain) NewCurrentHeadEVMCaller() (vm.EVMCaller, error) {
+	header := bc.CurrentHeader()
+	state, err := bc.StateAt(header.Root)
```

Comment:
This looks like the same race condition we identified in the EVMCaller PR

Reply:
This is not, but LightChain does have it.

here you get the header, and the you get the state from the header.

the race is getting "current" header, and then getting "current" state. Where the race is that maybe the notion of "current" has changed

Reply:
Ahh, yes thanks for pointing that out, what about the light chain though?

Reply:
Suggested fix for lightchain at https://github.com/celo-org/celo-blockchain/pull/1522/files#r632842994 .  Marking this one as resolved.